src/WakuRlnV2.sol

```diff
@@ -133,34 +135,65 @@ contract WakuRlnV2 is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         return rateCommitment != 0;
     }
 
-    /// Allows a user to register as a member
+    /// @notice Allows a user to register as a member
     /// @param idCommitment The idCommitment of the member
     /// @param userMessageLimit The message limit of the member
+    function register(uint256 idCommitment, uint32 userMessageLimit) external onlyValidIdCommitment(idCommitment) {
+        if (memberExists(idCommitment)) revert DuplicateIdCommitment();
+
+        uint32 index;
+        bool reusedIndex;
+        (index, reusedIndex) = _acquireMembership(_msgSender(), idCommitment, userMessageLimit, true);
+
+        _register(idCommitment, userMessageLimit, index, reusedIndex);
+    }
+
+    /// @notice Allows a user to register as a member
+    /// @param idCommitment The idCommitment of the member
+    /// @param userMessageLimit The message limit of the member
+    /// @param membershipsToErase List of expired idCommitments to erase
     function register(
         uint256 idCommitment,
-        uint32 userMessageLimit
+        uint32 userMessageLimit,
+        uint256[] calldata membershipsToErase
     )
         external
         onlyValidIdCommitment(idCommitment)
-        onlyValidUserMessageLimit(userMessageLimit)
     {
-        _register(idCommitment, userMessageLimit);
+        if (memberExists(idCommitment)) revert DuplicateIdCommitment();
+
+        for (uint256 i = 0; i < membershipsToErase.length; i++) {
+            uint256 idCommitmentToErase = membershipsToErase[i];
+            MembershipInfo memory mdetails = members[idCommitmentToErase];
```

Comment:
> Otherwise it looks like this will be implicitly checked through things like grace period and expiry, correct?

Hm. Good point. It's probably better to be explicit in this case.